### ISSUE TITLE
Self Host Public Server List

### ISSUE THREAD

**Issue by trymeouteh** [mumble-voip/mumble#5939]
### Description

The ability to self host your own public server list which can be used to index Mumble servers.

https://github.com/mumble-voip/mumble/discussions/5905
https://github.com/mumble-voip/mumble/discussions/5906

### Mumble component

Both

### OS-specific?

No

**Comment by Krzmbrzl** [comment]
What good will this do other than potentially fragment the server list?

**Comment by trymeouteh** [comment]
- It will decentralised mumble more with allowing multiple public server lists instead of one central public server list
- Mumble server with this feature, can have their server listed on 1, 2, 3 or multiple server lists. Not just be limited to one server list which will not fragment the server list.
- People can host a more private server list.

**Comment by Krzmbrzl** [comment]
Fair enough. I don't expect this to take any priority at this point though. But if anyone wants to give this a shot, let me know!
It will also require us to e.g. transform our service infrastructure into a Dockerfile, but that'd be a good thing anyway...

**Comment by davidebeatrici** [comment]
Well, a Dockerfile would be complementary.

The idea is to rewrite the infrastructure in a high performance language that can provide us most of the stuff we need in its standard library.

Lua has been serving us well, but we are relying on quite a few third-party components and there are some quirks in its syntax, such as indexes starting from 1 instead of 0.

I've been experimenting with [V](https://vlang.io) for the past ~2 months and I'd say it's ideal for something like this.

Basically, we would end up with a server that can then be used through a reverse proxy (Nginx usually).
